contracts/stellar-axelar-gateway/src/auth.rs

```diff
@@ -100,23 +70,20 @@ pub fn rotate_signers(
 
     let new_signers_hash = new_signers.hash(env);
 
-    let new_epoch: u64 = epoch(env) + 1;
+    let new_epoch = storage::epoch(env) + 1;
 
-    env.storage().instance().set(&DataKey::Epoch, &new_epoch);
+    storage::set_epoch(env, &new_epoch);
 
-    env.storage()
-        .persistent()
-        .set(&DataKey::SignersHashByEpoch(new_epoch), &new_signers_hash);
+    storage::set_signers_hash_by_epoch(env, new_epoch, &new_signers_hash);
 
     ensure!(
-        epoch_by_signers_hash(env, new_signers_hash.clone()).is_err(),
+        storage::try_epoch_by_signers_hash(env, new_signers_hash.clone())
+            .ok_or(ContractError::InvalidSignersHash)
+            .is_err(),
```

Comment:
```suggestion
            .is_none(),
```
check the Option directly